crates/indexer/src/accountsdb/accounts/candy_machine.rs

```diff
@@ -0,0 +1,249 @@
+use indexer_core::{
+    db::{
+        custom_types::{
+            EndSettingType as DbEndSettingType, WhitelistMintMode as DbWhitelistMintMode,
+        },
+        insert_into,
+        models::{
+            CMCollectionPDA, CMConfigLine, CMCreator, CMEndSetting, CMGateKeeperConfig,
+            CMHiddenSetting, CMWhitelistMintSetting, CandyMachine as DbCandyMachine,
+            CandyMachineData as CMData,
+        },
+        tables::{
+            candy_machine_collection_pdas, candy_machine_config_lines, candy_machine_creators,
+            candy_machine_datas, candy_machine_end_settings, candy_machine_gate_keeper_configs,
+            candy_machine_hidden_settings, candy_machine_whitelist_mint_settings, candy_machines,
+        },
+    },
+    prelude::*,
+};
+use mpl_candy_machine::{
+    CandyMachine, CollectionPDA, ConfigLine, EndSettingType, WhitelistMintMode,
+};
+
+use super::Client;
+use crate::prelude::*;
+
+pub(crate) async fn process(
+    client: &Client,
+    key: Pubkey,
+    candy_machine: CandyMachine,
+) -> Result<()> {
+    let cm = DbCandyMachine {
+        address: Owned(bs58::encode(key).into_string()),
+        authority: Owned(bs58::encode(candy_machine.authority).into_string()),
+        wallet: Owned(bs58::encode(candy_machine.wallet).into_string()),
+        token_mint: candy_machine
+            .token_mint
+            .map(|t| Owned(bs58::encode(t).into_string())),
+        items_redeemed: candy_machine.items_redeemed.try_into()?,
+    };
+
+    client
+        .db()
+        .run(move |db| {
+            insert_into(candy_machines::table)
+                .values(&cm)
+                .on_conflict(candy_machines::address)
+                .do_update()
+                .set(&cm)
+                .execute(db)
+        })
+        .await
+        .context("failed to insert candy machine")?;
+
+    let cm_data = CMData {
+        candy_machine_address: Owned(bs58::encode(key).into_string()),
+        uuid: Owned(candy_machine.data.uuid),
+        price: candy_machine.data.price.try_into()?,
+        symbol: Owned(candy_machine.data.symbol),
+        seller_fee_basis_points: candy_machine.data.seller_fee_basis_points.try_into()?,
+        max_supply: candy_machine.data.max_supply.try_into()?,
+        is_mutable: candy_machine.data.is_mutable,
+        retain_authority: candy_machine.data.retain_authority,
+        go_live_date: candy_machine.data.go_live_date,
+        items_available: candy_machine.data.items_available.try_into()?,
+    };
+
+    client
+        .db()
+        .run(move |db| {
+            insert_into(candy_machine_datas::table)
+                .values(&cm_data)
+                .on_conflict(candy_machine_datas::candy_machine_address)
+                .do_update()
+                .set(&cm_data)
+                .execute(db)
+        })
+        .await
+        .context("failed to insert candy machine data")?;
+
+    for creator in candy_machine.data.creators.iter() {
```

Comment:
```suggestion
    for creator in candy_machine.data.creators {
```
Alternatively, you may need:
```suggestion
    for creator in &candy_machine.data.creators {
```